contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/edrt/optimizer/EDrtVehicleDataEntryFactory.java

```diff
@@ -74,7 +74,11 @@ public VehicleEntry create(DvrpVehicle vehicle, double currentTime) {
 		if (schedule.getStatus() == ScheduleStatus.PLANNED) {
 			nextTaskIdx = 0;
 			chargeBeforeNextTask = battery.getCharge();
-		} else { // STARTED
+			// STARTED, but the vehicle is out of service time
+		} else if (currentTime >= vehicle.getServiceEndTime()) {
```

Comment:
Consider changing this series of `if`s to a switch statement with cases for PLANNED, STARTED and default.

Reply:
I'm not so familiar with those `ScheduleStatus`, if a started state ensures also a vehicle is in operational time window? Maybe you might refactor this? Till this, I would go without switches.